Bugfixes

veos-sxarr opened the pull request:
We fixed bugs we found during test.
We also updated make files and spec files to create aveo package for VH and aveorun for VE in order to create debuginfo packages.

veos-sxarr committed:
Updated test programs to specify -1 to veo_proc_create().
Then, we can specify VE node number using environment variable VE_NODE_NUMBER.

veos-sxarr committed:
Fix a problem that result is not queued when submit function fails.

veos-sxarr committed:
Modified veo_proc_create_static() to return 0 when it catchs VEOException.
Modified the cause of VEOException() when urpc_wait_peer_attach() fails

veos-sxarr committed:
Update GettingStarted and Restriction.
 + Add "The Required Number of HugePages for VEO" to GettingStarted.md
 + Add some restrictions to Restriction.md

veos-sxarr committed:
Modified a problem that shared memory segment remains if urpc_wait_peer_attach() fails.

veos-sxarr committed:
Updated make files and spec files to create aveo package for VH and aveorun package for VE

Updated spec files to generate debuginfo packages

Updated make files to search libraris in build directories and install directories specified by variables

   VE: -Wl,-rpath,$(VEDEST)/lib -L$(BVELIB) -L$(PREF)$(VEDEST)/lib
   VH: -Wl,-rpath,$(DEST)/lib64 -L$(BLIB) -L$(PREF)$(DEST)/lib64

You can build test programs using pre-installed aveo packages as follows

   make DEST=/opt/nec/ve/veos VEDEST=/opt/nec/ve -C test

veos-sxarr committed:
Updated the tutorial to describe migration to AVEO from VEO

veos-sxarr committed:
Updated requires in spec files

veos-sxarr committed:
Updated Makefile to add RPMBUILD_DIR to specify the directory to build rpms

veos-sxarr committed:
Updated spec files to remove "Obsolute" in order to release the packages as a preview release

veos-sxarr committed:
Fixed an issue where VEO_COMMAND_OK is stored in the result when an
exception is occurred by calling veo_call_async_write().

veos-sxarr committed:
Updated Makefile to link aveorun with liburpcVE.a
Updated relink_aveorun.in to link aveorun with liburpcVE.a even if -fopenmp is specified

veos-sxarr committed:
Updated test_omp.c and libvetestomp.c to check the number of threads

veos-sxarr committed:
Fixed an issue where the created thread context is placed on core 0
if the process/thread already exists.

veos-sxarr committed:
Merge branch 'master' of https://github.com/SX-Aurora/aveo into bugfixes

Conflicts:
	src/Makefile

veos-sxarr committed:
Fixed the issue after merging

veos-sxarr committed:
Fixed make_aveo.inc to add a space in LDFLAGS

efocht commented on doc/Restriction.md:
```diff
@@ -1,5 +1,22 @@
- - to use quadruple precision real number a variable length character string 
-   as a return value and an argument of Fortran subroutines and functions,
+VEO does not support to execute VEO program without setting the number of HugePages more than 32.
+VEO does not support to use quadruple precision real number a variable length character string as a return value and an argument of Fortran subroutines and functions,
```
fixup language a bit later

veos-sxarr replied:
We pushed the patch to fix language.

efocht commented on src/Makefile:
```diff
@@ -107,25 +102,23 @@ $(BLIB)/libveo.a: $(VHLIB_OBJS) | $$(@D)/
 		< $< > $@
 	chmod 755 $@
 
+$(BLIBEX)/mk_veorun_static: ../scripts/relink_aveorun.in
```
Do we need twice the same script? IMO relink_aveorun is a more appropriate name and we should keep it and symlink the old name for compatibility.

veos-sxarr replied:
We pushed the patch to add relink_aveorun to aveorun package.
We made mk_veorun_static a symbolic link.

efocht commented on doc/GettingStarted.md.in:
```diff
@@ -3,60 +3,60 @@ VE Offloading framework (VEO) is a framework to provide accelerator-style
 
-veoffload-veorun is automatically installed by dependency.
+To develop programs, @PACKAGE@-devel and @PACKAGE@run-devel and
+thedevelopment packages of the compiler (2.2.2 or later) are also
```
typo, space missing

veos-sxarr replied:
Typo was already fixed in commit :4497f91.

https://github.com/veos-sxarr-NEC/aveo/commit/4497f91f4063eacb2b4bc22382490f3d64caf0de

efocht commented on scripts/relink_aveorun.in:
```diff
@@ -39,7 +39,7 @@ while [ $# -gt 0 ]; do
       ;;
     '-fopenmp')
-        default_LDFLAGS="${default_LDFLAGS} ${option} @velibdir@/liburpcVE_omp.a"
+        default_LDFLAGS="${default_LDFLAGS} ${option} @velibdir@/liburpcVE.a"
```
I don't understand this. When -fopenmp is selected in relink_aveorun, we want openmp!

veos-sxarr replied:
We already reverted this change.

https://github.com/SX-Aurora/aveo/pull/5/commits/48267ff3d55e1427ad27d4acba0fc16aa0fdfc3f

Currently, liburpcVE_omp.a is specified.

~~~
     '-fopenmp')
      default_LDFLAGS="${default_LDFLAGS} ${option} $VELIBDIR/liburpcVE_omp.a"
      is_openmp=1`
~~~

veos-sxarr committed:
Fixed language in Restriction.md
Removed an unnecessary description from Restriction.md

veos-sxarr committed:
Added relink_aveorun to aveorun package

Made mk_veorun_static a symblic link

Updated relink_aveorun to support the following environment variables in order to build test programs using binaries in a build directory
    VELIBDIR
    VELIBEXECDIR

veos-sxarr committed:
Updated GettingStarted.md.in to describe users can link their program with AVEO after installing package

veos-sxarr committed:
This reverts commit 51728171a8ed2a0cc48f3d5b70b15ca280f91c9f.
